### ISSUE TITLE
`optionalParams` not being destructured properly when using the RxJS operator or the `.logLevel` method directly

### ISSUE THREAD

**Issue by jits** [dewizz/ngx-toolkit#32]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
</code></pre>

## Packages

<pre><code>
[ ] @ngx-toolkit/cookie
[ ] @ngx-toolkit/device
[x] @ngx-toolkit/logger
[ ] @ngx-toolkit/spring
</code></pre>

## Current behavior

When using the RxJS operator, or the `.logLevel` method directly, the extra `optionalParams` are not being destructured properly before being passed on to relevant methods. [Example line](https://github.com/dewizz/ngx-toolkit/blob/04e887078446a6d713ad32b2697f7c487fd3aa9b/projects/logger/src/lib/logger.service.ts#L45).

This results in anything passed as `optionalParams` being wrapped in an unnecessary array in the output.

## Expected behavior

Avoid an unnecessary wrapping of extra optional params into an array when output to the console.

## Minimal reproduction of the problem with instructions

Example:

```js
loggerService.logLevel(Level.INFO, 'Test:', { data: { foo: 'bar' } });
```

… gives:

![Screenshot 2020-06-10 15 40 06](https://user-images.githubusercontent.com/31973/84282610-e0b61d00-ab31-11ea-8638-09dfc78744c0.png)

## What is the motivation / use case for changing the behavior?

Making the logs clearer.

I believe this is easy to fix if you destructure the `optionalParams` before passing it in to the method. Example fix for one of the lines:

```js
this.error(message, ...optionalParams);
```

I'm happy to make a PR if you agree.

## Environment

<pre><code>
Angular version: 9.1.9
Ngx-Toolkit version: 9.0.3

Browser:
- [x] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 12.16.1  
- Platform:  Mac 

Others:

</code></pre>

Thank you for your work on this library!

**Comment by npetillon** [comment]
Hi,
Thank you for your interest and the detailed explanations.
I fixed the issue in the PR #33  and a new release is published: v9.0.4

Don't hesitate to publish a PR next time ;)
And if you have any suggestions, new features, or others... you are welcome :)

**Comment by jits** [comment]
> I fixed the issue in the PR #33 and a new release is published: v9.0.4

Excellent! Thank you for the quick response and fix.

> Don't hesitate to publish a PR next time ;)

Sure thing, will do next time. Wanted to make sure this wasn't an intentional design decision before making any changes :)

> And if you have any suggestions, new features, or others... you are welcome :)

Will do. I've been looking for the "ultimate" Angular logging library and this comes very close :) May have more suggestions/feedback as I use it.
